### ISSUE TITLE
What is `SourceNS` the source of?

### ISSUE THREAD

**Issue by dafeda** [equinor/ert#2805]
I think it would be helpful to add a bit of context to `SourceNS`.
I can't tell how "stochastic", "storage", and "resources" are related.
Also, does NS stand for name space or something else?

https://github.com/equinor/ert/blob/dc2a04d491b32bd8e8a51c8a02740d4f606c10d2/ert3/config/_ensemble_config.py#L16-L19

**Comment by jondequinor** [comment]
`source` in the `EnsembleConfig` is just a string, because the input source would always be sampled from a _stochastic_ distribution. So you'd say

```python
source: coefficients
record: coefficients
```

and that was that (_not 100% certain of the history here, but doesn't matter_).

But then we introduced multiple sources from where inputs originate. For instance, if I had previously uploaded something to _storage_, I should be able to use that as a source. Or, if I had a file lying around inside the `resources` folder in my workspace, that could also be a source.

Instead making `source` a URI, we came up with this scheme instead: `<namespace>.<location>`

- `stochastic.coefficients`: a sampling based on the parameter configuration
- `storage.coefficients`: a record called `coefficients` is retrieved from storage
- `resources.coefficients.json`: a file located in `resources/coefficients.json` under the workspace.

`SourceNS` is _source namespace_, an enumeration of the three sources available to the user.

But this scheme isn't nearly as expressive as it ought to be, so maybe we could maybe turn this issue into how instead to fix this? For example, `storage` is ambiguous, and the assumption that `resources` is the only directory we need to consider, isn't a good one.

Maybe we need to support something like this

- atle_j@storage://192.168.0.1/doggerbank/FOTPZOPT
- file:///private/jond/A:/GJ�A/wolfenstein3d.exe
- stochastic://coefficients

It's not user friendly, but we could infer a lot of this.
